CONTRIBUTING.md

```diff
@@ -127,6 +127,25 @@ The conventional commit type (in PR title) is very important to automatically bu
 
 There is no need to update the CHANGELOG in a PR because it will be updated as part of the release process (see [RELEASING.md](RELEASING.md) for more details).
 
+### Testing
+
+Most unit tests case be run via:
+
+```sh
+npm test
+```
+
+However, some instrumentations require some test-services to be running (e.g. a MongoDB server to test the `instrumentation-mongodb` package). Use the `test-services`-related npm scripts to start test services in Docker and to run the tests with the appropriate configuration to use them:
+
+```sh
+npm run test-services:start     # starts services in Docker
+npm run test:with-test-services # runs 'npm test' with envvars from test/test-services.env
```

Comment:
is this command more about telling to use the env variables from the file you mentioned or does it also check if the required services are running?
Just thinking if make sense to combine this, and only call `run test:with-test-services` which will start the service and use the correspondent vars

Reply:
`npm run test:with-test-services` just sets the envvars and then runs the tests.


> Just thinking if make sense to combine this, and only call `run test:with-test-services` which will start the service and use the correspondent vars

The value in having `test-services:start` and `test:with-test-services` be separate commands is to support a dev/test cycle where one is repeatedly running the tests. It would be a time suck to start and stop services for each test run.

Or, if you are proposing that this new `npm run test:and-start-services-if-necessary` thing (it wouldn't have to be that awful long name :) would only start services if necessary and then run tests then:
- Yes, that would avoid the time burden of starting and stopping test services for each run.
- However, it creates an slight DX imbalance in that one still needs another script to stop any started test-services. Perhaps it is just a personal preference thing. I like the matching of `test-services:start` and `test-services:end`.
- Also, FWIW, `npm run test-services:{start,end}` support arguments do the underlying `docker compose ...` call. So, if one is just working on the redis tests, then one can `npm run test-services:start -- redis`. This is relatively minor though.

Reply:
Though I agree that without reading the developer guide docs that `npm run test:with-test-services` *does* potentially give the impression that it'll handle starting the test services for the developer. `npm run test:and-assume-test-services-are-running-cuz-imma-use-them` is a little long.

Reply:
I was indeed thinking of keeping the end separate. It was just that the `start` seems to be a requirement for the `with-test-services`, so it would make sense to have them as one, since you would always run them together. I didn't think about the case where you would `start` but not necessarily use the `with-test-services` after, so make sense to have the start separate.
It would be nice if the `with-test-services` could start the service in case you have forgotten to start, but that could be some future improvement. I would hold any approval based on this 😄

Reply:
`npm run test:just-run-all-the-things`